app/models/registration.rb

```diff
@@ -48,8 +48,9 @@ class Registration < ApplicationRecord
   end
 
   validates :guests, numericality: { greater_than_or_equal_to: 0 }
-
-  validates :guests, numericality: { less_than_or_equal_to: :guest_limit, if: :check_guest_limit? }
+  validates :guests, numericality: { less_than_or_equal_to: :guest_limit, if: :check_guest_limit?, frontend_code: Registrations::ErrorCodes::GUEST_LIMIT_EXCEEDED }
+  validates :guests, numericality: { equal_to: 0, unless: :guests_allowed?, frontend_code: Registrations::ErrorCodes::GUEST_LIMIT_EXCEEDED }
+  validates :guests, numericality: { less_than_or_equal_to: Registrations::RegistrationChecker::DEFAULT_GUEST_LIMIT, if: :guests_unrestricted?, frontend_code: Registrations::ErrorCodes::UNREASONABLE_GUEST_COUNT }
```

Comment:
We should probably work more with comments explaining the "business logic" idea behind each validation. But in principle, this is how you would pass information around.